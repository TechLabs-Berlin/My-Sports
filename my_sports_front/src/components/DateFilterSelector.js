import React, { useState } from 'react';
import { connect } from "react-redux";
import DatePicker from "react-datepicker";
import { filterByDate } from '../actions/filter';
import moment from 'moment';

const mapStateToProps = ({ filters }) => ({
    filters
  });

const mapDispatchToProps = dispatch => ({
    filterByDate: (payload) => dispatch(filterByDate(payload))
});

const DateFilterSelector = ({ filterByDate, filters }) => {

    const [startDate, setStartDate] = useState(new Date());

    const filterDate = (date) => {
        setStartDate(date)
        filterByDate(moment(date).format('YYYY-MM-DD'));
    }

    const removeDateFilter = () => {
        setStartDate(new Date())
        filterByDate('')
    }

    return (
        <div className="accordion-item">
            <h2 className="accordion-header filter-type-selector" id="flush-headingDate">
            <button className="accordion-button collapsed" type="button" data-bs-toggle="collapse" data-bs-target="#flush-collapseDate" aria-expanded="false" aria-controls="flush-collapseDate">
                By Date {filters.date && <span className="mx-auto ">{moment(filters.date).format('D MMM')}</span>}
            </button>
            </h2>
            <div id="flush-collapseDate" className="accordion-collapse collapse" aria-labelledby="flush-headingDate" data-bs-parent="#accordionFlushExample">
            <div className="accordion-body">
                <DatePicker
                selected={startDate}
                onChange={date => filterDate(date)}
                inline
                />
                <p><button className="remove-date-filter-button" onClick={removeDateFilter}>Remove Date Filter</button></p>
            </div>
            </div>
        </div>
    )
}


export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DateFilterSelector);