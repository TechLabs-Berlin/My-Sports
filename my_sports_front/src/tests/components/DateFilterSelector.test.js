import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import moment from "moment";
import DateFilterSelector from "../../components/DateFilterSelector";
import { filters } from "../fixtures/filters";
import "../setupTests";

let startDate, wrapper;
const filterByDate = {
  type: "FILTER_BY_DATE",
  payload: {},
};

beforeEach(() => {
  startDate = jest.fn();
  wrapper = shallow(
    <Provider store={store}>
      <DateFilterSelector filterByDate={filterByDate} filters />
    </Provider>
  ).dive();
});

// test("should render date filter selector correctly", () => {
//   expect(wrapper).toMatchSnapshot();
// });

// test('should set start date to default moment', () => {

// });
