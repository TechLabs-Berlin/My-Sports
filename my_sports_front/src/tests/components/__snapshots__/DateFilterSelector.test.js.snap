// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render date filter selector correctly 1`] = `undefined`;
